#18570 Chart Indicator label positioning

arunasr opened the pull request:
With default options, the label is clipped at the start or at the end, depending on positioning, unless offset is specified, in which case one has to manually calculate the offset; the labels still look bad because they are rendered aligned to the middle.
This patch makes the labels hug the left or the right margin respectively, and they auto-size nicely.

dylans commented:
You'll want to rebase this pull request against master, as it's including the changes for dojox/lang/functional also.

arunasr commented:
@dylans I am unfamiliar with the procedures, can you please nudge me in the right direction what precisely shall I do. Point me to the doc or just give me the commands to use. I have precious little time to filter the correct information from the mass of the guidelines available.

wkeese commented:
I think in your case you'd want to do something like:

```
$ git checkout master
$ git pull origin   (to get latest changes from dojox repository in the dojo account)
$ git checkout f20150428
$ git rebase master
$ git push --force arunsar f20150428   (to push the update to your github account)
```

arunasr commented:
Well, thanks a lot for a try, but no joy:

arunasr@fuji:~/github/dojox$*git checkout master*

Switched to branch 'master'

Your branch is up-to-date with 'origin/master'.

arunasr@fuji:~/github/dojox$*git pull origin*

Already up-to-date.

arunasr@fuji:~/github/dojox$*git checkout f20150428*

Switched to branch 'f20150428'

Your branch is up-to-date with 'origin/f20150428'.

arunasr@fuji:~/github/dojox$*git rebase master*

Current branch f20150428 is up to date.

arunasr@fuji:~/github/dojox$*git status*

On branch f20150428

Your branch is up-to-date with 'origin/f20150428'.

nothing to commit, working directory clean

arunasr@fuji:~/github/dojox$*git remote -v*

origin    git@github.com:arunasr/dojox.git (fetch)

origin    git@github.com:arunasr/dojox.git (push)

upstream    git@github.com:dojo/dojox.git (fetch)

upstream    git@github.com:dojo/dojox.git (push)

arunasr@fuji:~/github/dojox$*git remote show origin*

* remote origin

   Fetch URL: git@github.com:arunasr/dojox.git

   Push  URL: git@github.com:arunasr/dojox.git

   HEAD branch: master

   Remote branches:

     1.0       tracked

     1.1       tracked

     1.10      tracked

     1.2       tracked

     1.3       tracked

     1.4       tracked

     1.5       tracked

     1.6       tracked

     1.7       tracked

     1.8       tracked

     1.9       tracked

     f20150428 tracked

     master    tracked

   Local branches configured for 'git pull':

     f20150428 merges with remote f20150428

     master    merges with remote master

   Local refs configured for 'git push':

     f20150428 pushes to f20150428 (up to date)

     master    pushes to master    (up to date)

arunasr@fuji:~/github/dojox$*git push --force origin f20150428*

Everything up-to-date

dylans commented:
Ok, then I suggest a new branch and new pull request with the change, since it's just one line anyway.

Also, please let me know which test(s) you've run to verify this so I can quickly understand the change in behavior.

arunasr commented:
Pull request #176 submitted.
Good point about testing. Before I tested only "vertical:false" 
(start:true and false), these worked well.
After testing "vertical=true" I added the condition to centre the 
vertical labels on the line.
Unfortunately labels at the top of the chart are still clipped at the 
top. I do not know how to fix them. I only need horizontal indicator lines.

neonstalwart commented:
for anyone reading this... the assumption was that dojo's repo was at origin but it's at upstream so the instructions at https://github.com/dojo/dojox/pull/175#issuecomment-97421730 should be:

```
$ git checkout master
$ git pull upstream   (to get latest changes from dojox repository in the dojo account)
$ git checkout f20150428
$ git rebase master
$ git push --force origin f20150428   (to push the update to your github account)
```

arunasr commented:
@neonstalwart Thanks a lot Ben, one day it will make perfect sense to me. Just not today... :)

arunasr commented:
@neonstalwart That did the trick, with one modification: `git pull upstream master` was the second command that I needed.

dylans commented:
btw, I think your original issue @arunasr might be that you were using the same branch for multiple pull requests... it's usually easier to use a separate branch for each.

dylans commented:
Thanks for the fix, closed via 265915b .

arunasr commented:
@dylans Thanks, please ignore/delete pull request #176 then.
Yes I used the same branch for this. The reason is that I need all my modifications simultaneously for my project to work. If I have one modification per branch, I can have only one modification at a time. I know there must be a way to have yet another private branch with all my modifications merged, but this is too much to fit in my brain at this time.
